app/controllers/upload_controller.go

```diff
@@ -0,0 +1,96 @@
+package controllers
+
+import (
+	"encoding/csv"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/tealeg/xlsx"
+	"github.com/webonise/csv_upload/pkg/framework"
+)
+
+func (s *Srv) RenderUploadView(w *framework.Response, r *framework.Request) {
+	tmplList := []string{
+		"./web/views/upload.html",
+		"./web/layouts/flash.html",
+	}
+	res, err := s.TplParser.ParseTemplate(tmplList, nil)
+	if err != nil {
+		s.Log.Error(err)
+		http.Error(w.ResponseWriter, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.RenderHTML(res)
+}
+
+func (s *Srv) UploadFile(w *framework.Response, r *framework.Request) {
+
+	var returnMessage string
+
+	file, handler, err := r.FormFile("file")
+	if err != nil {
+		returnMessage = "Error while retrieving uploaded file"
+	}
+	defer file.Close()
+	//Extracting the extension of file
+	fslice := strings.Split(handler.Filename, ".")
+	fileExtension := fslice[len(fslice)-1]
+	//Handling CSV and XLSX file extension
+	switch fileExtension {
+	case "csv": //Handle file with extension CSV
+		reader := csv.NewReader(file)
+		record, _ := reader.ReadAll()
```

Comment:
why ignored error?

Reply:
okay i will change code to handle this error